grakn-graph/src/main/java/ai/grakn/graph/internal/TypeImpl.java

```diff
@@ -118,7 +135,23 @@ public void innerDelete(){
         if(hasSubs || hasInstances){
             throw new ConceptException(ErrorMessage.CANNOT_DELETE.getMessage(getName()));
         } else {
+            //Force load of linked concepts whose caches need to be updated
+            cachedSuperType.get();
+            cachedPlaysRoles.get();
+
             deleteNode();
+
+            //Update neighbouring caches
+            //noinspection unchecked
+            ((TypeImpl<T, V>) cachedSuperType.get()).deleteCachedDirectedSubType(getThis());
+            cachedPlaysRoles.get().forEach(roleType -> ((RoleTypeImpl) roleType).deleteCachedDirectPlaysByType(getThis()));
```

Comment:
Nope again. You won't have any subtypes if you get to this stage. This check: `boolean hasSubs = getVertex().edges(Direction.IN, Schema.EdgeLabel.SUB.getLabel()).hasNext();` will make sure of that.